src/blueteam/Gardener.java

```diff
@@ -137,30 +165,137 @@ boolean build(RobotType type) {
 		}
 	}
 
-	@Override void step() {
-		// TODO Lukas
+	/**
+	 * assuming were in garden so build direction is NOT random
+	 *
+	 * @param type of robot
+	 * @return if successful
+	 */
+	boolean build(RobotType type) {
+		return build(type, false);
+	}
+
+	/**
+	 * checks whether enemy Archon is nearby,
+	 * used in init phase
+	 *
+	 * @return true if is near
+	 */
+	boolean isEnemyArchonNear() {
+		RobotInfo[] nearbyRobots = rc.senseNearbyRobots();
+		for (RobotInfo robot : nearbyRobots) {
+			if (robot.getType() == RobotType.ARCHON && robot.getTeam() == enemy)
+				return true;
+		}
+		return false;
+	}
 
+	/**
+	 * Checks whether there is a lot (const) trees around
+	 *
+	 * @return true if is more than TeamConstants.GARDENER_NUM_OF_TREES_TO_BUILD_LUMBER
+	 */
+	boolean isInWoods() {
+		TreeInfo[] nearbyTrees = rc.senseNearbyTrees(TeamConstants.GARDENERS_DEFAULT_FREE_SPOT_RADIUS, Team.NEUTRAL);
+		System.out.println(nearbyTrees.length);
+		if (nearbyTrees.length > TeamConstants.GARDENER_NUM_OF_TREES_TO_BUILD_LUMBER)
+			return true;
+		else
+			return false;
+	}
+	@Override void dodge()
+	{
+
+	}
```

Comment:
I understand that dodging inside of garden is pretty useless. Maybe if you are in the process of finding garden it could be useful to not get killed easily.